Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/TotScatCalculateSelfScattering.py

```diff
@@ -75,10 +79,19 @@ def PyExec(self):
                                           BinningForCalc=[min_x, 1 * width_x, max_x],
                                           BinningForFit=[min_x, 10 * width_x, max_x],
                                           FitSpectrumWith="CubicSpline")
-        self_scattering_correction = CalculatePlaczekSelfScattering(InputWorkspace=raw_ws,
-                                                                    IncidentSpectra=fit_spectra,
-                                                                    ScaleByPackingFraction=False,
-                                                                    Version=1)
+
+        plackzek_order = self.getPropertyValue('PlaczekOrder')
+        placzek_kwargs = {'InputWorkspace': raw_ws, 'IncidentSpectra': fit_spectra, 'ScalebyPackingFraction': False,
+                          'Order': plackzek_order}
+        if plackzek_order == '2':
+            sample_temp = self.getPropertyValue('SampleTemp')
+            try:
+                if float(sample_temp):
+                    placzek_kwargs.update({'SampleTemperature': float(sample_temp)})
+            except ValueError:
+                raise RuntimeError("To calculate the second order Placzek correction you must provide "
+                                   "a sample_temp.")
```

Comment:
The CalculatePlaczek algorithm checks whether a temperature has been supplied for order=2 so I'm wondering if we need to do a validation here as well or whether we can just leave it to the CalculatePlaczek algorithm? The validation in the algorithm also checks if a temperature is present in the sample logs.

If you do leave this validation code in place perhaps this should be "...you must provide a sample temperature" rather than using the variable name.